internal/utils_common/files_test.go

```diff
@@ -19,6 +20,130 @@ type fileDetail struct {
+		}
+		deletedCount++
+	}
+	return deletedCount, nil
+}
+
+// MockFileRemover is a mock implementation of FileRemover.
+type MockFileRemover struct {
+	RemovedPaths []string
+}
+
+// RemoveAll mocks the RemoveAll method.
+func (m *MockFileRemover) RemoveAll(path string) error {
+	m.RemovedPaths = append(m.RemovedPaths, path)
+	return nil
+}
+
+func Test_deletePaths(t *testing.T) {
+	// Create temporary files and directories for testing
+	dir, err := ioutil.TempDir("", "testdir")
+	if err != nil {
+		t.Fatalf("Failed to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(dir) // Clean up
+
+	file1 := filepath.Join(dir, "file1.txt")
+	file2 := filepath.Join(dir, "file2.txt")
+
+	_, err = os.Create(file1)
+	if err != nil {
+		t.Fatalf("Failed to create temporary file: %v", err)
+	}
+	_, err = os.Create(file2)
+	if err != nil {
+		t.Fatalf("Failed to create temporary file: %v", err)
+	}
+
+	paths := []string{file1, file2}
+
+	// Call the function under test
+	deletedCount, err := deletePaths(paths)
+	if err != nil {
+		t.Errorf("deletePaths returned an unexpected error: %v", err)
+	}
+
+	// Assert that the number of deleted paths matches the expected count
+	expectedDeletedCount := len(paths)
+	if deletedCount != expectedDeletedCount {
+		t.Errorf("deletePaths deleted %d paths, expected %d", deletedCount, expectedDeletedCount)
+	}
+
+	// Assert that the paths were deleted
+	for _, path := range paths {
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
+			t.Errorf("deletePaths did not remove path: %s", path)
+		}
+	}
+}
+
+func TestCopyOptions_Validate(t *testing.T) {
+	// Set up the test cases with input data and expected validation results
+	testCases := []struct {
+		InputData     *CopyOptions
+		ExpectedError string
+		TestCaseID    int
+	}{
+		// Test case 1: Valid input data
+		{
+			InputData: &CopyOptions{
+				Source:      "valid_source",
+				Destination: "valid_destination",
+			},
+			ExpectedError: "", // Expect no validation errors
+			TestCaseID:    1,
+		},
+		// Test case 2: Missing source
+		{
+			InputData: &CopyOptions{
+				Source:      "",
+				Destination: "valid_destination",
+			},
+			ExpectedError: "'source' must have a value",
+			TestCaseID:    2,
+		},
+		// Test case 3: Missing destination
+		{
+			InputData: &CopyOptions{
+				Source:      "valid_source",
+				Destination: "",
+			},
+			ExpectedError: "'destination' must have a value",
+			TestCaseID:    3,
+		},
+		// Add more test cases as needed
+	}
+
+	// Loop through the test cases
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("TestCaseID=%d", tc.TestCaseID), func(t *testing.T) {
+			// Call the Validate method with the test data
+			err := tc.InputData.Validate()
+
+			// Assert that the error message matches the expected error message
+			if tc.ExpectedError != "" {
+				require.Error(t, err)
+				assert.EqualError(t, err, tc.ExpectedError)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
```

Comment:
you can update this using the pattern we discussed about yesterday, where you have **name** (so you won't need to comment on what the code does), and **assertion**

Reply:
ok gaw on it